addons/pos_restaurant/models/pos_session.py

```diff
@@ -56,8 +56,11 @@ def get_pos_ui_restaurant_floor(self):
 
     def _load_onboarding_data(self):
         super()._load_onboarding_data()
+        restaurant = self.env.ref('pos_restaurant.pos_config_main_restaurant', raise_if_not_found=False)
+        if not restaurant:
+            return
```

Comment:
```suggestion
        restaurant = self.env.ref('pos_restaurant.pos_config_main_restaurant', raise_if_not_found=False) or self.env['pos.config'[
```

Don't return early because there is procedure that needs to be run even if `restaurant` is `None`.